qfieldcloud_sdk/sdk.py

```diff
@@ -85,29 +153,64 @@ class OrganizationMemberModel(TypedDict):
 
 
 class Pagination:
+    """The Pagination class allows for controlling and managing pagination of results within the QFieldCloud SDK.
+
+    Attributes:
+        limit (Optional[int]): The maximum number of items to return.
+        offset (Optional[int]): The starting point from which to return items.
+    """
+
     limit = None
     offset = None
 
     def __init__(
         self, limit: Optional[int] = None, offset: Optional[int] = None
     ) -> None:
+        """Initializes the pagination settings.
+
+        Args:
+            limit (Optional[int]): The maximum number of items to return. Defaults to None.
+            offset (Optional[int]): The starting point from which to return items. Defaults to None.
+        """
         self.limit = limit
         self.offset = offset
 
     @property
     def is_empty(self):
+        """Checks if both limit and offset are None, indicating no pagination settings.
+
+        Returns:
+            bool: True if both limit and offset are None, False otherwise.
+        """
         return self.limit is None and self.offset is None
 
 
 class Client:
+    """The core component of the QFieldCloud SDK, providing methods for interacting with the QFieldCloud platform.
+
+    This class handles authentication, project management, file management, and more.
+
+    Attributes:
+        session (requests.Session): The session object to maintain connections.
+        url (str): The base URL for the QFieldCloud API.
+        token (str): The authentication token for API access.
+        verify_ssl (bool): Whether to verify SSL certificates.
+    """
```

Comment:
Do we want to come up with a rule for class attributes (i.e. should they all be declared above __init__ with default values? Less risks of creating situations where an attribute isn't initialized in __init__ and cause an error?

Reply:
The thing is that sometimes these attributes are not nullable but also cannot have defaults that make sense. What I would suggest is we added them below like this:

```suggestion
    Attributes:
        session (requests.Session): The session object to maintain connections.
        url (str): The base URL for the QFieldCloud API.
        token (str): The authentication token for API access.
        verify_ssl (bool): Whether to verify SSL certificates.
    """

    session: requests.Session

    url: str

    token:str

    veryfy_ssl: bool
```